### PR TITLE
add rwmb_set_meta

### PR DESCRIPTION
**Pull request by hsimah**
#1366 raised to look into the possibility of this

Would love to hear thoughts and feedback on naming and how I am doing things. This is mostly pulled from the REST API plugin. I am not sure of the safety of calling `rwmb_get_storage` where I am but I don't know the codebase well enough.

### CODE REVIEW

**Comment by rilwis** [comment]
Thanks for your suggestion. This is a good idea!

I think there are 2 issues with the current code:

- `$post_id` should be changed to `$object_id`, and is a mandatory field. So, it's better to put it on the first of the argument list
- And we should consider for other object type, not just post.

**Comment by hsimah** [comment]
> * `$post_id` should be changed to `$object_id`, and is a mandatory field. So, it's better to put it on the first of the argument list
Fixed.
> * And we should consider for other object type, not just post.
Ah yes - when I am querying a User I submit `['object_type' => 'user']` as the `$args`. I assume checking that and falling back to `get_post_type` is okay?

Are there any other types I should be aware of? I don't use WordPress much.

File: inc/functions.php [hunk 1]
```diff
@@ -33,6 +33,51 @@ function rwmb_meta( $key, $args = array(), $post_id = null ) {
...
+
+		if ( false === $field ) {
+			$args  = wp_parse_args(
+				$args,
+				array(
+					'type'     => 'text',
+					'multiple' => false,
+					'clone'    => false,
+				)
+			);
```
Comment (hsimah) [inline inc/functions.php:61]: Is this necessary? Can I put the default values in the first line of the function?

**Comment by rilwis** [comment]
Hi @hsimah 

Sorry for the delay. I've added some comments. Can you please update the code?

**Comment by hsimah** [comment]
@rilwis sorry for my delay, I got added as a collaborator on WPGraphQL and now my notifications on GitHub are swamped with their issues.

I don't see any comments on my PR? And you merged it too. If you have any further work you want me to do to finish this I am happy to do it :)

**Pull request merged**